filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReceiver.java

```diff
@@ -48,33 +140,34 @@ private void registerMethods() {
     // receiveFile after getting a serveFile method call). handler needs to implement receiveFile method
     private List<Method> receiveFileMethod(Object handler) {
         List<Method> methods = new ArrayList<>();
-        methods.add(new Method(RECEIVE_META_METHOD, "ssl", "ii", handler,"receiveFileMeta")
-                .paramDesc(0, "filereference", "file reference to download")
-                .paramDesc(1, "filename", "filename")
-                .paramDesc(2, "filelength", "length in bytes of file")
+        methods.add(new Method(RECEIVE_META_METHOD, "sssl", "ii", handler,"receiveFileMeta")
+                .paramDesc(0, "filereference", "inprogressFile reference to download")
+                .paramDesc(1, "type", "'inprogressFile' or 'compressed'")
+                .paramDesc(2, "filename", "filename")
+                .paramDesc(3, "filelength", "length in bytes of inprogressFile")
                 .returnDesc(0, "ret", "0 if success, 1 otherwise")
                 .returnDesc(1, "session-id", "Session id to be used for this transfer"));
         methods.add(new Method(RECEIVE_PART_METHOD, "siix", "i", handler,"receiveFilePart")
-                .paramDesc(0, "filereference", "file reference to download")
+                .paramDesc(0, "filereference", "inprogressFile reference to download")
```

Comment:
I guess you didn't mean to rename all 'file' to 'inprogressFile'

Reply:
Fixed